src/superpmi-collect/superpmicollect.cs

```diff
@@ -0,0 +1,777 @@
+
+            if (outputMchPath == null)
+            {
+                s_saveFinalMchFile = false;
+                s_finalMchFile = Path.Combine(s_tempDir, "final.mch");
+                s_tocFile = Path.Combine(s_tempDir, "final.mch.mct");
+            }
+            else
+            {
+                s_saveFinalMchFile = true;
+                s_finalMchFile = Path.GetFullPath(outputMchPath);
+                s_tocFile = s_finalMchFile + ".mct";
+            }
+        }
+
+        private static int RunProgram(string program, string arguments)
+        {
+            if (program.LastIndexOf(".py") != -1)
+            {
+                string programArgumentSep = String.IsNullOrEmpty(arguments) ? "" : " ";
+                arguments = "-u " + program; 
+                program = "python";
+            }
+
+            else if (program.LastIndexOf(".dll") != -1)
+            {
+               string programArgumentSep = String.IsNullOrEmpty(arguments) ? "" : " ";
+               arguments = program;
+               program = "dotnet";
+            }
+            
+            // If the program is a script, move the program name into the arguments, and run it
+            // under the appropriate shell.
+            if (Global.IsWindows)
+            {
+                if ((program.LastIndexOf(".bat") != -1) || (program.LastIndexOf(".cmd") != -1))
+                {
+                    string programArgumentSep = String.IsNullOrEmpty(arguments) ? "" : " ";
+                    arguments = "/c " + program + programArgumentSep + arguments;
+                    program = Environment.GetEnvironmentVariable("ComSpec"); // path to CMD.exe
+                }
+            }
+            else
+            {
+                if (program.LastIndexOf(".sh") != -1)
+                {
+                    string programArgumentSep = String.IsNullOrEmpty(arguments) ? "" : " ";
+                    arguments = "bash " + program + programArgumentSep + arguments;
+                    program = "/usr/bin/env";
+                }
+            }
+
+            Console.WriteLine("Running: " + program + " " + arguments);
+            Process p = Process.Start(program, arguments);
+            p.WaitForExit();
+            return p.ExitCode;
+        }
+
+        // Run a single test from the coreclr test binary drop.
+        // This works even if given a test path in Windows file system format (e.g.,
+        // "c:\foo\bar\runit.cmd") when run on Unix. It converts to Unix path format and replaces
+        // the ".cmd" with ".sh" before attempting to run the script.
+        private static void RunTest(string testName)
+        {
+            string testDir;
+
+            if (Global.IsWindows)
+            {
+                int lastIndex = testName.LastIndexOf("\\");
+                if (lastIndex == -1)
+                {
+                    throw new SpmiException("test path doesn't have any directory separators? " + testName);
```

Comment:
Is this basically testing to make sure that we are supplying full paths to the tests? Is that necessary?